host/frontend/host-orchestrator/operator.go

```diff
@@ -0,0 +1,400 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+//     Unless required by applicable law or agreed to in writing, software
+//     distributed under the License is distributed on an "AS IS" BASIS,
+//     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//     See the License for the specific language governing permissions and
+//     limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/gorilla/mux"
+)
+
+func main() {
+	pool := NewDevicePool()
+	polledSet := NewPolledSet()
+	var config interface{}
+	err := json.Unmarshal([]byte("{\"ice_servers\":[{\"urls\":[\"stun:stun.l.google.com:19302\"]}],\"message_type\":\"config\"}"), &config)
+	if err != nil {
+		log.Fatal("Malformed config string")
+	}
+
+	setupDeviceEndpoint(pool, config)
+	r := setupServerRoutes(pool, polledSet, config)
+
+	http.Handle("/", r)
+	if err := http.ListenAndServe(":8080", nil); err != nil {
+		log.Fatal("ListenAndServe client: ", err)
+	}
+}
+
+func setupDeviceEndpoint(pool *DevicePool, config interface{}) {
+	devMux := http.NewServeMux()
+	if devMux == nil {
+		log.Fatal("Failed to allocate ServeMux")
+	}
+	// http.HandleFunc("/", serveHome)
+	devMux.HandleFunc("/register_device", func(w http.ResponseWriter, r *http.Request) {
+		deviceWs(w, r, pool, config)
+	})
+	// Serve the register_device endpoint in a different thread
+	go func() {
+		err := http.ListenAndServe("127.0.0.1:8081", devMux)
```

Comment:
Listening on this address enforces that it only accepts local connections, right?

Reply:
Yes, as opposed to ":8080" which accepts anything.